backtester/eventhandlers/portfolio/holdings/holding.go

```diff
@@ -0,0 +1,90 @@
+package holdings
+
+import (
+	"errors"
+
+	"github.com/thrasher-corp/gocryptotrader/backtester/common"
+	"github.com/thrasher-corp/gocryptotrader/backtester/eventtypes/fill"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/order"
+)
+
+// Create takes a fill event and creates a new holding for the exchange, asset, pair
+func Create(f fill.Event, initialFunds, riskFreeRate float64) (Holding, error) {
+	if f == nil {
+		return Holding{}, errors.New("nil event received")
+	}
+	if initialFunds <= 0 {
+		return Holding{}, errors.New("initial funds <= 0")
+	}
+	h := Holding{
+		Pair:           f.Pair(),
+		Asset:          f.GetAssetType(),
+		Exchange:       f.GetExchange(),
+		Timestamp:      f.GetTime(),
+		InitialFunds:   initialFunds,
+		RemainingFunds: initialFunds,
+		RiskFreeRate:   riskFreeRate,
+	}
+	h.update(f)
+
+	return h, nil
+}
+
+// Update calculates holding statistics for the events time
+func (h *Holding) Update(f fill.Event) {
+	h.Timestamp = f.GetTime()
+	h.update(f)
+}
+
+// UpdateValue calculates the holding's value for a data event's time and price
+func (h *Holding) UpdateValue(d common.DataEventHandler) {
+	h.Timestamp = d.GetTime()
+	latest := d.Price()
+	h.updateValue(latest)
+}
+
+func (h *Holding) update(f fill.Event) {
+	direction := f.GetDirection()
+	o := f.GetOrder()
+	switch direction {
+	case order.Buy:
+		h.PositionsSize += o.Amount
+		h.PositionsValue += o.Amount * o.Price
+		h.RemainingFunds -= (o.Amount * o.Price) + o.Fee
+		h.TotalFees += o.Fee
+		h.BoughtAmount += o.Amount
+		h.BoughtValue += o.Amount * o.Price
+	case order.Sell:
+		h.PositionsSize -= o.Amount
+		h.PositionsValue -= o.Amount * o.Price
+		h.RemainingFunds += (o.Amount * o.Price) - o.Fee
+		h.TotalFees += o.Fee
+		h.SoldAmount += o.Amount
+		h.SoldValue += o.Amount * o.Price
+	case common.DoNothing, common.CouldNotSell, common.CouldNotBuy, common.MissingData, "":
+	}
+	h.TotalValueLostToSlippage += (f.GetVolumeAdjustedPrice() - f.GetPurchasePrice()) * f.GetAmount()
+	h.TotalValueLostToVolumeSizing += (f.GetClosePrice() - f.GetVolumeAdjustedPrice()) * f.GetAmount()
+	h.updateValue(f.GetClosePrice())
+}
+
+func (h *Holding) updateValue(l float64) {
+	origPosValue := h.PositionsValue
+	origBoughtValue := h.PositionsValue
```

Comment:
are origPosValue and origBoughtValue supposed to be the same? in that case why do we need both

Reply:
No, but you did pick up a typo! 
# 🦖